libs/agno/agno/workflow/workflow.py

```diff
@@ -345,6 +298,66 @@ async def result_generator():
             logger.warning(f"Workflow.arun() should only return RunResponse objects, got: {type(result)}")
             return None
 
+    async def arun_workflow_generator(self, **kwargs: Any) -> AsyncIterator[RunResponse]:
+        """Run the Workflow asynchronously for async generators"""
+
+        # Set mode, debug, workflow_id, session_id, initialize memory
+        self.set_storage_mode()
+        self.set_debug()
+        self.set_monitoring()
+        self.set_workflow_id()  # Ensure workflow_id is set
+        self.set_session_id()
+        self.initialize_memory()
+
+        # Create a run_id
+        self.run_id = str(uuid4())
+
+        # Set run_input, run_response
+        self.run_input = kwargs
+        self.run_response = RunResponse(run_id=self.run_id, session_id=self.session_id, workflow_id=self.workflow_id)
+
+        # Read existing session from storage
+        self.read_from_storage()
+
+        # Update the session_id for all Agent instances
+        self.update_agent_session_ids()
+
+        log_debug(f"Workflow Run Start: {self.run_id}", center=True)
+        # Initialize the run_response content
+        self.run_response.content = ""
+        try:
+            self._subclass_run = cast(Callable, self._subclass_run)
+            async for item in self._subclass_run(**kwargs):
+                if (
+                    isinstance(item, tuple(get_args(RunResponseEvent)))
+                    or isinstance(item, tuple(get_args(TeamRunResponseEvent)))
+                    or isinstance(item, tuple(get_args(WorkflowRunResponseEvent)))
+                    or isinstance(item, RunResponse)
+                ):
+                    # Update the run_id, session_id and workflow_id of the RunResponseEvent
+                    item.run_id = self.run_id
+                    item.session_id = self.session_id
+                    item.workflow_id = self.workflow_id
```

Comment:
Unrelated to the PR but why are we setting these ids after the subclass runs and not when starting the runs?

Reply:
we're just updating those values on the event itself.